Automattic/pocket-casts-android
# End of Year: Listened categories design

| 📘 Project: #410 |
|:---:|

iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/453

## Description
Adds the design for the "listened categories" story.

* The podcasts cover displayed are determined by getting the list of top categories and reversing it, picking the top 1 podcast for each of them
* Background is picked from the middle podcast in the stack (without any gradient)

## Testing Instructions
1. Set `END_OF_YEAR_ENABLED` feature flag to `true` in `base.gradle`
2. Run the app and make sure you're logged in to an account with a few items on Listening History
3. When the prompt appears, tap "View My 2022"
4. Navigate to the third story
4. ✅ Check the design

## Screenshots or Screencast 

| Phone (Pixel 5) |
| -------- | 
|  <img width = 320 src="https://user-images.githubusercontent.com/1405144/201099562-1486674c-935d-4471-9155-d7680e0eba87.png"/>|

| Tablet (Nexus 9) |
| -------- | 
|<img width = 500 src="https://user-images.githubusercontent.com/1405144/201099600-8a0cc476-6b36-418e-97df-5c2de4498c3c.png"/> |

P.S. -> Story speed issue has re-surfaced after `StoriesFragment` (bottom sheet fragment) was removed in favor of Compose dialog in [this commit](https://github.com/Automattic/pocket-casts-android/commit/07478904b503f0b7718ff120a72d920f28a6e4d4). I tried to fix in 67bf509e11297d02b28c549b851dc78f67a692d2 but I can still the issue. I'll try to fix it in the next PR.

## Checklist
- \If this is a user-facing change, I have added an entry in CHANGELOG.md
- I have considered whether it makes sense to add tests for my changes
- Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...
